### DISCUSSION TITLE
Eigenvalue executioner

### DISCUSSION THREAD

**Discussion by tannhorn** [arfc/moltres#239] [Q&A Software development] [answered]
Hello, 

I have been testing if I could get Moltres running with the latest Moose framework. However, it tells me 

_'NonlinearEigen' executioner is deprecated in favor of 'Eigenvalue' executioner._

when I execute the PinCell input file. The 'Eigenvalue' executioner, however, requires Problem type to be an 'EigenProblem' or a derived subclass. When I try to set that instead of a 'FEProblem', I simply get 'segmentation fault' as the only output without further explanation. Any pointers on why this happens and how this could be remedied? Thanks!

**Comment by smpark7** [comment, marked as answer]
Hi @Tannhorn, thanks for trying out Moltres!

`NonlinearEigen` is still fully functional, as are all other deprecated items. `Eigenvalue` relies on different interface parameters to set up the eigenproblem. I have the necessary changes ready to go in an unmerged PR (#228), but we're waiting on the MOOSE team to iron out some performance issues with the new `Eigenvalue` executioner (see #197).

To summarize, `NonlinearEigen` and other deprecated objects are all still fully functional. Feel free to let us know of any other deprecated objects so that we'll put it on our agenda for the next Moltres code updates.

**Reply by tannhorn** [reply]
Thank you for the quick answer!
